horovod/common/operations.cc

```diff
@@ -260,6 +260,11 @@ const Status SHUT_DOWN_ERROR = Status::Aborted(
     "exception, you should see the exception in the log before the first "
     "shutdown message.");
 
+const Status DUPLICATE_NAME_ERROR = Status::InvalidArgument(
+    "Requested to allreduce, allgather or broadcast a tensor with a same name "
+    "as another tensor that is currently being processed.  Please make sure "
+    "this is a desired behavior and if so, use a different tensor name.");
```

Comment:
A little confused about this message.
Maybe just say "If you want to request another tensor, use a different tensor name."